### ISSUE TITLE
Changing notebook-dir while jupyter lab is still running

### ISSUE THREAD

**Issue by kavishdahekar** [jupyterlab/jupyterlab#6204]
Assuming I start jupyter lab with the --notebook-dir set to a specific folder.

Is there anyway to change this directory during runtime without restarting jupyter lab?

**Comment by jasongrout** [comment]
I don't think so. I think this is a Jupyter notebook server constraint, so it may be better to ask in a Jupyter notebook repo issue.

**Comment by kavishdahekar** [comment]
@jasongrout I see.
Is there anyway I start individual jupyter notebooks in different directories from jupyterlab?

**Comment by lresende** [comment]
@kavishdahekar What are you trying to accomplish? Would subdirectories work for your scenario?

**Comment by kavishdahekar** [comment]
@lresende this is what I want to accomplish.
Assume the following directory structure:

```
root
  |---notebooks
           |---project1
           |      |---notebook1.ipynb
           |---project2
                  |---notebook2.ipynb

```

Assume my jupyter lab instance is already running with ```--notebook-dir = "root"```

Now, the user has direct links to notebook1 and notebook2. For example:
http://localhost:8888/lab/tree/project1/notebook1.ipynb
http://localhost:8888/lab/tree/project2/notebook2.ipynb

When the user clicks on the link to notebook1.ipynb, I want him to be able to view only "project1" directory in the file browser widget and he should not be able to navigate to "project2" or "root" directory. Same for when he is using notebook2.ipynb, he should be able to access only "project2" directory.

In essence, I want project1 and project2 folders and their notebooks to be isolated from each other in the file browser.

I thought, if in the **same jupyter lab instance**, if I could somehow start notebook1.ipynb with ```--notebook-dir = root/project1``` folder and notebook2.ipynb with ```--notebook-dir = root/project2```, then the problem would be solved.

Any ideas how to achieve this please?

**Comment by jasongrout** [comment]
> Any ideas how to achieve this please?

Not very easily. I think the generally accepted way of doing this is starting up a separate notebook server for each project.
